parser/src/lexer.rs

```diff
@@ -74,6 +75,26 @@ pub enum LexicalErrorType {
     OtherError(String),
 }
 
+impl fmt::Display for LexicalError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self.error {
+            LexicalErrorType::StringError => write!(f, "Got unexpected string"),
+            LexicalErrorType::UnicodeError => write!(f, "Got unexpected unicode"),
+            LexicalErrorType::NestingError => write!(f, "Got unexpected nesting"),
+            LexicalErrorType::UnrecognizedToken { tok } => {
+                write!(f, "Got unexpected token {}", tok)
+            }
+            LexicalErrorType::OtherError(ref msg) => write!(f, "{}", msg),
+        }?;
+        write!(
```

Comment:
I would prefer to add the line and column info later on, so in the `Display` method of the `CompilerError` struct.

Reply:
Done